### PR TITLE
623: add JSX.Element type to supporting prop definition

### PR DESCRIPTION
**Pull request by joecowton1**
## What is the purpose of this change?

As per https://github.com/guardian/source/issues/623 this PR addresses the request for a `supportingMedia` prop on the `RadioGroup` component.

This is preferred to adding an additional type to the `supporting` prop ie `string | ReactNode` as `supporting` is used as a text input to other components where it makes no sense to pass a React component. 

**I've tried to stick to Source conventions where I've noticed them, but apologies if I've missed anything! Please pull me up on anything like that.

## What does this change?

-   add `supportingMedia` prop
-   add corresponding stories

## Screenshots

Screenshot 2021-06-24 at 16 57 25

### Accessibility

-   [x] [Tested with screen reader](https://www.theguardian.design/2a1e5182b/p/6691bb-accessibility/t/78ac51)
-   [x] [Navigable with keyboard](https://www.theguardian.design/2a1e5182b/p/6691bb-accessibility/t/009027)
-   [x] [Colour contrast passed](https://www.theguardian.design/2a1e5182b/p/6691bb-accessibility/t/58dbf2)
-   [x] [The component doesn't use only colour to convey meaning](https://www.theguardian.design/2a1e5182b/p/6691bb-accessibility/t/61524d)

### Cross browser and device testing

-   [ ] Tested with touch screen device

### Responsiveness

-   [ ] Tested at all breakpoints
-   [ ] Tested with with long text
-   [ ] Stretched to fill a wide container

### Documentation

-   [x] Full API surface area is documented in the README
-   [x] Examples in Storybook

### CODE REVIEW

**Comment by SiAdcock** [comment]
Thanks so much @joecowton1 

On this:

> This is preferred to adding an additional type to the supporting prop ie string | ReactNode as supporting is used as a text input to other components where it makes no sense to pass a React component.

I think you're referring to the `supporting` prop on the `Legend` component. Do you not think the Legend component's supporting prop should also be a `string | ReactNode`?

**Comment by joecowton1** [comment]
> Thanks so much @joecowton1
> 
> On this:
> 
> > This is preferred to adding an additional type to the supporting prop ie string | ReactNode as supporting is used as a text input to other components where it makes no sense to pass a React component.
> 
> I think you're referring to the `supporting` prop on the `Legend` component. Do you not think the Legend component's supporting prop should also be a `string | ReactNode`?

@SiAdcock I think you're right on this, I've changed the approach just to use the `supporting` prop for both text and media components.

**Comment by SiAdcock** [comment]
Thanks @joecowton1 this is looking great! Can I ask for a couple of things?

- Add a story for the change to the Legend component
- Update the Legend docs to say that it now supports any JSX element
- Split the changes to Legend and the changes to RadioGroup into separate commits. This helps makes the changelog more coherent!

thanks so much for your help so far 🙏

File: src/core/components/radio/index.tsx [hunk 1]
```diff
@@ -127,7 +127,7 @@ interface RadioProps extends InputHTMLAttributes<HTMLInputElement>, Props {
 	checked?: boolean;
 	defaultChecked?: boolean;
 	label?: ReactNode;
-	supporting?: ReactNode;
+	supporting?: ReactNode | JSX.Element;
```
Comment (SiAdcock) [inline src/core/components/radio/index.tsx:130]: Is this intentional? I wonder what is the difference between these types, in a practical sense?

Comment (joecowton1) [reply src/core/components/radio/index.tsx:130]: No that's just me not paying attention. Reverted :)

**Comment by joecowton1** [comment]
@SiAdcock I _think_ I've covered all those tasks now, please say if I've missed anything!

**Review by SiAdcock — APPROVED** [review]
Thanks so much @joecowton1!

**Pull request merged**